### DISCUSSION TITLE
Trigger in FieldArray causes form re-renders

### DISCUSSION THREAD

**Discussion by christianJessMark** [react-hook-form/react-hook-form#6332] [Q&A]
I have a complex form with mainly controlled inputs, since we need to do a lot of cross-field validations. 

Some of the fields requires validation on change, therefore we use the `trigger` API. 

However using that (in a nested fieldArray) causes our entire form to re-render on each stroke, which causes laggy behaviour on the input. We've so far debounced the trigger, but seems the entire form should not re-render?

**Comment by bluebill1049** [comment]
which version are you on? and where do you subscribed to your formState: errors? isValid?

**Reply by christianJessMark** [reply]
`7.13.0-next.5`

Subscribing to `errors`

**Reply by bluebill1049** [reply]
and you are subscribed at the root right? perhaps you want to use `useFormState`

**Comment by christianJessMark** [comment]
Yes, root 

```
  const {
    register,
    formState: { errors },
  } = useFormContext()
  ```
  
 I can try the `useFormState` hook - is it also possibly to drill into my field array errors?

**Reply by christianJessMark** [reply]
When I try to set the form `mode` to `onChange` - I see the re-renders are isolated to the field array. So that has better performance, rather than triggering validation for a single field.

**Comment by christianJessMark** [comment]
It looks like the `name` in `useFormState` doesn't work correctly. 

This is a pretty ugly implementation, but I've tried subscribing just to the field array in this forked sandbox, and still I also see the root errors. 

https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-forked-28rjb?file=/src/fieldArray.js:617-681

**Reply by bluebill1049** [reply]
can you provide more detail on it? steps to reproduce?

**Reply by christianJessMark** [reply]
I'll create a nicer example and open an issue ✌️
